### PR TITLE
Adds ahelp jobbans

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
:cl: Kor
add: Admins can now ban people from admin helping, to discourage frivolous and malicious ahelps. Admin helps are not meant as a method of OOC revenge when you've been obnoxious IC, so please don't use them as such.
/:cl:

Why:

DEAD: John Sweeps says, "i can also ahelp constantly about it until i get someone sympathetic"
DEAD: John Sweeps says, "so even if 90% of admins don't care if I get that 10% that do"
DEAD: John Sweeps says, "they get baned and I win"

KyrahAbattoir: ahelp is free and there is a small chance to turn around the situation, why not use it basically anytime someone bests you. [To be clear Kyrah does not think this is a good thing they are not a bad player!]

I think we need a way to slap people on the wrist for ahelping after they instigated a fight and got killed, but admins rarely want to jump to a full server ban, so maybe this can be a middleground.

>WHAT IF THE GUY GETS GRIEFED FOR REAL WHILE HE IS AHELP BANNED

Then he shouldn't have cried wolf so much and wasted everyones time/tried to use the support line to grief people beforehand

### CODE REVIEW

**Comment by Jalleo** [comment]
This needs to be added alongside logging of how often people Ahelp and from what perspective they are in the round TBH.

**Comment by BeeSting12** [comment]
kor doing gods work here

**Comment by Hatfish** [comment]
Would it be possible to add an option to just ban someone from ahelping for the round?

**Comment by Dorsisdwarf** [comment]
@Hatfish you can already mute people from Ahelping

**Comment by Jordie0608** [comment]
I don't think this is a good idea.

**Comment by PKPenguin321** [comment]
I worry this will be misused if we're not really clear to adminbus about what it's for

**Comment by Shadowlight213** [comment]
I agree with Jordie. The most we should do is maybe have a way to flag someone as a serial ahelped or something so it notifies admins to check notes.
Ahelp muting is fine enough as it is. If it's consistent enough that muting isn't enough, a server ban is a better solution instead of blocking someone's main access to admins.

**Comment by KorPhaeron** [comment]
Every admin button in the game can be misused if the admins are intent on misusing it.

We have OOC and even emote bans as half measures before a full serverban, not sure why people are resistant to this one

**Comment by AdamElTablawy** [comment]
Truly, I am a hero.

**Comment by Cruix** [comment]
Might want to ban them from pray as well

**Comment by BeeSting12** [comment]
adam that doesnt make you a hero it makes you a shitter for being a banbaiting asshole

**Comment by Bawhoppen** [comment]
real problem: tg has bad players

File: code/modules/admin/verbs/adminhelp.dm [hunk 1]
```diff
@@ -458,6 +458,10 @@ GLOBAL_DATUM_INIT(ahelp_tickets, /datum/admin_help_tickets, new)
 	set category = "Admin"
 	set name = "Adminhelp"
 
+	if(jobban_isbanned(src, "AHELP"))
+		src << "<span class='danger'>You have been banned from adminhelping. You will have to appeal this ban on the forums.</span>"
```
Comment (lzimann) [inline code/modules/admin/verbs/adminhelp.dm]: `to_chat`

**Comment by Jordie0608** [comment]
@KorPhaeron Because OOC and emote bans don't prevent players from communicating to admins in what may actually be a situation where one is needed by the player. How does a player resolve such a situation if they can't ahelp? The ban requests forum?

**Comment by lzimann** [comment]
I feel like this is not the way to go. And banning people from ahelping might set a really bad precedent. Regardless, I doubt this will ever get used, since if a person gets to the point of being ahelp banned, he might aswell be just thrown out of the server completly.

**Comment by KorPhaeron** [comment]
>prevent players from communicating to admins in what may actually be a situation where one is needed by the player

Well that is the point of a punishment, there are adverse consequences

**Comment by Nilons** [comment]
There needs to be a specific criteria for when this is issued I think so that it's not just used on anyone who's wrong in an ahelp

**Comment by KorPhaeron** [comment]
I would personally suggest using this for

-People who leave out key information in ahelps ("CLOWN KILLED ME FNR...oh yeah I guess I did steal all his stuff earlier)

-People who cry in ahelps after they acted like an asshat all round (SECURITY GULAGGED ME AND ALL I DID WAS STEAL EVERYTHING IN MEDBAY SPACE LAW DOESNT LET THEM DO THAT!)

**Comment by Nilons** [comment]
That would solve the only problem I can see with this and honestly it's about time ban baiting shutters had a recourse against them.  I was worried cause I ahelp pretty much everything suspect because I don't give people the benefit of the doubt that they're traitor unless it's really obvious.  People who blatantly leave info out in hopes the admin won't investigate are cancer but not cancer enough for a server ban

**Comment by lzimann** [comment]
Both situations are already bannable. If a person is constantly doing these things will be server banned.

**Comment by KorPhaeron** [comment]
Why do we have jobbans then?

**Comment by Nilons** [comment]
@lzimann what's wrong with giving admins more options?

**Comment by Bawhoppen** [comment]
I see no real reason for this and at best it'll probably be abused by the chucklemins

**Comment by cedarbridge** [comment]
>I see no real reason for this and at best it'll probably be abused by the chucklemins

You have a certainly odd perspective of the nature and needs of administration for somebody who does not participate in said administration.

**Comment by Bawhoppen** [comment]
in what scenario would an ahelp ban be useful where a normal ban wouldn't also be applicable?

**Comment by Nilons** [comment]
It solve the problem of people fucking around in ahelped in a much neater way.  Why ban someone for lying in an ahelp when you can just stop them from ahelping?

**Comment by Shadowlight213** [comment]
Why not make attack bans next?
Why ban someone for non antag murder when you can stop them from attacking anyone?
Why ban them for shocking doors when you can ban them from interacting with anything?

**Comment by Bawhoppen** [comment]
@Nilons if you're limiting a potentially integral aspect of the game, they shouldn't be in the game in the first place

**Comment by Nilons** [comment]
Those examples don't work because they're all of bidness. By your logic we should server ban instead of ooc ban anyone, or server ban instead of job bans.  Ahelps are a privilege and admins should have the ability to take it away

**Comment by Nilons** [comment]
Because they're all ic bidness*. Autocorrect

**Comment by Niknakflak** [comment]
Censoring people from complaining is the first step to a total totalitarian admin government

**Comment by Bawhoppen** [comment]
we need to protect the people from themselves

**Comment by lzimann** [comment]
@tgstation/commit-access I don't think this is needed at all, so I guess merge or I'll close eventually

**Comment by Jordie0608** [comment]
I am as of yet, still unconvinced.

**Comment by KorPhaeron** [comment]
Take a poll of admins or something then, so far most of them seem to want it

File: code/modules/admin/verbs/adminhelp.dm (hunk 1 above)
Comment (ChangelingRain) [inline code/modules/admin/verbs/adminhelp.dm]: `to_chat`

**Comment by KorPhaeron** [comment]
@ChangelingRain 

done

**Comment by optimumtact** [comment]
Headmins couldn't decide so I'll say no on behalf of everyone everywhere

**Comment by dannno** [comment]
Tragic.

**Comment by KorPhaeron** [comment]
by imblyings » Tue Jun 13, 2017 3:42 pm #307850

fuck I failed to support the pr

I totally forgot

IM SORRY

**Comment by KorPhaeron** [comment]
@ausops

**Comment by Jordie0608** [comment]
I still think this is a bad idea =|

**Comment by lzimann** [comment]
It is rather pointless.

**Comment by kevinz000** [comment]
tbh just slap them wit ha mute for "lets spam ahelp" and a warning/ban if they cross the line of "oh im going to leave out critical information to make myself look in the right"/banbaiting.

**Comment by Armhulen** [comment]
When I was a player I was against it but some people really do need this, sadly

**Comment by Ausops** [comment]
I had a think about it. 

It's just more satisfying to watch and passively let someone gain momentum towards a rule 0 removal rather than a half-measure like an ahelp ban. I base this on two assumptions. One, that someone being truly awful and abusive with the intention of abusing ahelps to bother admins or hurt other players cannot genuinely change their character with just the usual slow escalation of warnings and bans by admins, at least not in a quick enough time frame. Two, that even if they were capable of such change, adminhelp bans do nothing to help that change that a rule 0 cannot, and do not address the likelihood of someone being this awful in ahelps also being awful in game.

In fact, abusing ahelps requires some prior knowledge of SS13 communities and just about no regard for other players, and the last thing I want or I suspect any ss13 server admin wants, is a player who maliciously uses the ahelp system to be on the server at all.

As for joke or bad meme ahelps these can just be disregarded and given the extinction treatment, which involves absolutely no attention given whatsoever, even in the form of negative attention via ahelp bans.

**Comment by Qbopper** [comment]
>it's just more satisfying to watch and passively let someone gain momentum towards a rule 0 removal

this is a fair and reasonable opinion, but when's the last time there was a successful rule 0 ban

I'm not trying to imply anything or comment I'm genuinely asking

**Comment by KorPhaeron** [comment]
We have a new term of headmins, and we still have issues with banbaiting, so I'd like a new vote on this with the next set

**Comment by KorPhaeron** [comment]
I vote yes

**Comment by Cyberboss** [comment]
I vote no

**Comment by LanCartwright** [comment]
ban them from adminwho with this.

**Comment by KorPhaeron** [comment]
I bet Cyberboss is the one who ate the floorpills and ahelped

**Comment by lzimann** [comment]
Ban the players instead of adding an useless ban.

**Comment by PKPenguin321** [comment]
This needs to be used on the poo guy

**Comment by Jordie0608** [comment]
Kor cmon at least wait until there actually are three new headmins before trying to shove this through again. You still have 2 weeks to go.

**Comment by KorPhaeron** [comment]
Thought it was just one week of voting

**Comment by Jordie0608** [comment]
That'd still be a week too prompt then nerd.

**Comment by optimumtact** [comment]
Projecting so hard you should change your name to KorImax

**Comment by Shadowlight213** [comment]
Sorry Kor, headmins have no say in the code.
oranges was very clear.

**Comment by Armhulen** [comment]
I support this pr, it has a niche use as a ban. Reopen when the new headmin term starts or something.

**Comment by Shadowlight213** [comment]
Once again, this is a maintainer decision, not a headmin one.
If we are actually having this be a headmin vote, then you have to let goof do code favors.
While I don't really feel strongly either way on this pr, I think saying one thing, and then going "oh it's ok when goof isn't doing it" is dishonest. (I never thought I would be defending goof)

**Comment by RemieRichards** [comment]
They have no say unless we query them, that's perfectly reasonable but obviously you wouldn't see it that way.

**Comment by KorPhaeron** [comment]
Do you honestly not see how not letting headmins boss around maintainers on game design and also voluntarily deferring to headmins on the inclusion/exclusion of admin tools is entirely consistent I assumed your first post was a joke

**Comment by Cyberboss** [comment]
Counselling != Directing

**Comment by optimumtact** [comment]
Yes lets do it

**Comment by lzimann** [comment]
let's not

**Comment by Bawhoppen** [comment]
I still don't get why anyone would want this other than to fuck with people. If someone is a shitter enough to be ahelp banned, then they're also a shitter enough for a regular ban.

Also, why are we calling it an ahelp jobban????

**Comment by Cyberboss** [comment]
Because it's implemented using the jobban system, which is just a text based key.

**Comment by Shadowlight213** [comment]
Alright, I see the point. It was just looking like it was being used as a way to bypass maintainer disapproval on a controversial pr. I guess I misunderstood.
No need for character attacks.

**Comment by Qbopper** [comment]
@Bawhoppen addressed in the OP

>I think we need a way to slap people on the wrist for ahelping after they instigated a fight and got killed, but admins rarely want to jump to a full server ban, so maybe this can be a middleground.

not taking a side here but kor did mention this

**Comment by Bawhoppen** [comment]
Yes I read that, but I don't think there are any circumstances where this could just be considered "a slap on the wrist". If someone is fucking around with ahelps and need just a slap on the wrist, you can always just mute them from it for the round.

**Comment by PKPenguin321** [comment]
>I still don't get why anyone would want this other than to fuck with people.

The poo guy

**Comment by optimumtact** [comment]
just ban them, nobody is gonna shed a tear

**Pull request closed without merging**